ros2doctor/ros2doctor/api/package.py

```diff
@@ -26,7 +28,12 @@
 from ros2doctor.api.format import doctor_error
 from ros2doctor.api.format import doctor_warn
 
-import rosdistro
+try:
+    import rosdistro
+except ImportError:
+    doctor_warn(
+        'Unable to import rosdistro. '
+        f'Use `{Path(sys.executable).name} -m pip install rosdistro` to install needed package.')
```

Comment:
I'm not really sure about this bit.  If `rosdistro` is a pip package that can be installed on Windows, then I feel like that should just be part of our installation instructions.  If it can't be installed on Windows, then I think we just want to skip this stuff.  Having this halfway place where we warn if certain things aren't installed seems weird to me.

Reply:
So, the way `ros2doctor` implements "optional" checks is odd. I'll ignore the fact that all dependencies are listed in the `package.xml`, which we generally assume to be satisfied. Why replicating checks everywhere, instead of failing entry point load with a meaningful message if dependencies are not met?

Reply:
> Why replicating checks everywhere, instead of failing entry point load with a meaningful message if dependencies are not met?

Yes, that's a good point.

See https://github.com/ros2/ros2cli/pull/631/commits/cd5e980710c595875bcf00d108b623da45e4d8c4 and https://github.com/ros2/ros2cli/pull/631/commits/fbf7510f248d57be25b07d5ef970d4742a96cccd.